File: tests/smoke_review_ingest.ts
#!/usr/bin/env node
/**
 * Smoke unitário do ingest (sem gh / rede).
 * Delega ao unit test de finding-ids + ingest.
 */
import { spawnSync } from "node:child_process";
import path from "node:path";
import { fileURLToPath } from "node:url";

const root = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "..");
const unit = path.join(
  root,
  "packages/code-review/tests/ingest/finding-ids-and-ingest.test.ts"
);
const tsx = path.join(root, "node_modules/.bin/tsx");

const proc = spawnSync(tsx, ["--test", unit], {
  cwd: root,
  encoding: "utf8",
  stdio: ["ignore", "pipe", "pipe"],
});
if (proc.stdout) process.stdout.write(proc.stdout);
if (proc.stderr) process.stderr.write(proc.stderr);
if (proc.status === 0) {
  console.log("smoke_review_ingest: OK");
}
process.exit(proc.status ?? 1);

File: tests/smoke_review_ingest.py
#!/usr/bin/env python3
"""Smoke unitário do ingest (sem gh / rede).

Delega ao unittest de lib; mantém entrypoint curto para npm test / smoke.
Uso: python3 tests/smoke_review_ingest.py
"""
from __future__ import annotations

import subprocess
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
UNIT = ROOT / "packages" / "code-review" / "tools" / "tests" / "test_ingest_and_ids.py"


def main() -> int:
    proc = subprocess.run(
        [sys.executable, str(UNIT)],
        cwd=str(ROOT),
        check=False,
    )
    if proc.returncode == 0:
        print("smoke_review_ingest: OK")
    return proc.returncode


if __name__ == "__main__":
    raise SystemExit(main())
